src/Compiler/Driver/CreateILModule.fs

```diff
@@ -54,11 +54,12 @@ module AttributeHelpers =
             | Some(Attrib(_, _, [ AttribBoolArg p ], _, _, _, _)) -> Some p
             | _ -> None
 
+    [<return: Struct>]
     let (|ILVersion|_|) (versionString: string) =
```

Comment:
These are covered in @psfinaki PR likely, and we decided to postpone changes for now, since we already have some SO reports in compiler and those need careful verification.

Reply:
https://github.com/dotnet/fsharp/pull/16822

Reply:
Reason to postpone it is here: https://github.com/dotnet/fsharp/pull/16316#issuecomment-1847252952
There were stackoverflows when allocating more things on stack in certain cases. We will need to make sure we aren't hitting (or won't be hitting) it with AP changes.